janus_core/calculations/phonons.py

```diff
@@ -0,0 +1,421 @@
+        phonon.run_mesh()
+        self.results["phonon"] = phonon
+
+        if self.logger:
+            self.logger.info("Phonons calculation complete")
+
+        if write_results:
+            self.write_phonons()
+
+    def write_phonons(
+        self,
+        *,
+        params_file: Optional[PathLike] = None,
+        force_const_file: Optional[PathLike] = None,
+        bands_file: Optional[PathLike] = None,
+    ) -> None:
+        """
+        Write results of phonon calculations.
+
+        Parameters
+        ----------
+        params_file : Optional[PathLike]
+            Name of yaml file to save phonon parameters. Default is inferred from
+            `file_prefix`.
+        force_const_file : Optional[PathLike]
+            Name of hdf5 file to save force constants. Default is inferred from
+            `file_prefix`.
+        bands_file : Optional[PathLike]
+            Name of yaml file to save band structure. Default is inferred from
+            `file_prefix`.
+        """
+        params_file = params_file if params_file else f"{self.file_prefix}-params.yml"
+        bands_file = bands_file if bands_file else f"{self.file_prefix}-auto_band.yml"
+        if not force_const_file:
+            force_const_file = f"{self.file_prefix}-force_consts.hdf5"
+
+        phonon = self.results["phonon"]
+
+        phonon.save(params_file, settings={"force_constants": False})
+        write_force_constants_to_hdf5(phonon.force_constants, filename=force_const_file)
+        phonon.auto_band_structure(write_yaml=True, filename=bands_file)
+
+    def calc_thermal_props(self, write_results: bool = True) -> None:
+        """
+        Calculate thermal properties and optionally write results.
+
+        Parameters
+        ----------
+        write_results : bool
+            Whether to write out results to file. Default is True.
+        """
+        # Calculate phonons is not already run
+        if "phonon" not in self.results:
+            self.calc_phonons(write_results=False)
+
+        if self.logger:
+            self.logger.info("Beginning thermal properties calculation")
+
+        self.results["phonon"].run_thermal_properties(
+            t_step=self.t_step, t_max=self.t_max, t_min=self.t_min
+        )
+        self.results["thermal_properties"] = self.results[
+            "phonon"
+        ].get_thermal_properties_dict()
+
+        if self.logger:
+            self.logger.info("Thermal properties calculation complete")
+
+        if write_results:
+            self.write_thermal_props()
+
+    def write_thermal_props(self, filename: Optional[PathLike] = None) -> None:
+        """
+        Write results of thermal properties calculations.
+
+        Parameters
+        ----------
+        filename : Optional[PathLike]
+            Name of data file to save thermal properties. Default is inferred from
+            `file_prefix`.
+        """
+        filename = filename if filename else f"{self.file_prefix}-cv.dat"
```

Comment:
This is such a common pattern, we may want a function somewhere which is something like:
```python
def _default_filename(self, filename: Optional[PathLike], default: str):
    if filename:
        return filename
    return f"{self.file_prefix}-{default}"
```

Reply:
True, this may solve the consistency issue too.